src/components/info/templates/template-utilities.ts

```diff
@@ -0,0 +1,152 @@
+  prefix: string
+): {
+  key: string
+  value: any
+}[] {
+  return Object.entries(attributes)
+    .filter(([key]) => key !== 'showProfile')
+    .map(([key, value]) => ({ key: prefix + key, value }))
+}
+
+export function hasAttributes(feature: FeatureJSON): boolean {
+  return (
+    feature.attributes !== undefined &&
+    Object.keys(feature.attributes).length > 0
+  )
+}
+
+export function isEmpty(value: string | undefined | null): boolean {
+  return value === undefined || value === null || value.length === 0
+}
+
+export function isLink(value: any): boolean {
+  const strValue = String(value).toLowerCase()
+  return strValue.startsWith('http://') || strValue.startsWith('https://')
+}
+
+export function showAttributesByLang(
+  elem: { [key: string]: any },
+  layerid: string,
+  language: string
+): boolean {
+  const { key } = elem
+  let ok = false
+
+  if (
+    layerid === '2407' &&
+    (key === 'f_Cl_erosion' ||
+      key === 'f_Erosion_kl' ||
+      key === 'f_Erosion_Cl' ||
+      key === 'f_Erosioun_K')
+  ) {
+    if (language === 'fr' && key === 'f_Cl_erosion') {
+      ok = true
+    } else if (language === 'de' && key === 'f_Erosion_kl') {
+      ok = true
+    } else if (language === 'en' && key === 'f_Erosion_Cl') {
+      ok = true
+    } else if (language === 'lb' && key === 'f_Erosioun_K') {
+      ok = true
+    }
+  } else if (
+    layerid === '2714' &&
+    (key === 'f_LABEL_eng' ||
+      key === 'f_LC_class_name_fr' ||
+      key === 'f_LC_class_name_de' ||
+      key === 'f_LC_class_name')
+  ) {
+    if (language === 'fr' && key === 'f_LC_class_name_fr') {
+      ok = true
+    } else if (language === 'de' && key === 'f_LC_class_name_de') {
+      ok = true
+    } else if (language === 'en' && key === 'f_LABEL_eng') {
+      ok = true
+    } else if (language === 'lb' && key === 'f_LC_class_name') {
+      ok = true
+    }
+  } else if (
+    layerid === '2444' &&
+    (key === 'f_Betriebsform' ||
+      key === 'f_Forme_d_entreprise' ||
+      key === 'f_Operation_Form' ||
+      key === 'f_Form_vun_Geschaeft')
+  ) {
+    if (language === 'fr' && key === 'f_Forme_d_entreprise') {
+      ok = true
+    } else if (language === 'de' && key === 'f_Betriebsform') {
+      ok = true
+    } else if (language === 'en' && key === 'f_Operation_Form') {
+      ok = true
+    } else if (language === 'lb' && key === 'f_Form_vun_Geschaeft') {
+      ok = true
+    }
+  } else if (
+    key.endsWith('_FR') ||
+    key.endsWith('_DE') ||
+    key.endsWith('_EN') ||
+    key.endsWith('_LU') ||
+    key.endsWith('_LB')
+  ) {
+    if (language === 'fr' && key.endsWith('_FR')) {
+      ok = true
+    } else if (language === 'de' && key.endsWith('_DE')) {
+      ok = true
+    } else if (language === 'en' && key.endsWith('_EN')) {
+      ok = true
+    } else if (
+      language === 'lb' &&
+      (key.endsWith('_LB') || key.endsWith('_LU'))
+    ) {
+      ok = true
+    }
+  } else {
+    ok = true
+  }
+
+  return ok
```

Comment:
If you have time it oculd be great to refactor this huge test into small ones with named cases (eg. `storage-rules-version-heper.ts`)

Reply:
True, this could be refactored into something much shorter I think, I've left it as is for now due to limited time.